solr/core/src/java/org/apache/solr/cli/SolrCLI.java

```diff
@@ -445,8 +450,31 @@ private static void printHelp() {
     print("Pass -help or -h after any COMMAND to see command-specific usage information,");
     print("such as:    ./solr start -help or ./solr stop -h");
   }
+
+  /**
+   * Strips off the end of solrUrl any /solr when a legacy solrUrl like http://localhost:8983/solr
+   * is used, and warns those users. In the future we'll have url's with /api as well.
+   *
+   * @param solrUrl with /solr stripped off.
+   * @return the truncated if need solrUrl.
+   */
+  public static String resolveSolrUrl(String solrUrl) {
+    if (solrUrl != null) {
+      if (solrUrl.indexOf("/solr") > -1) { //
+        solrUrl = solrUrl.substring(0, solrUrl.indexOf("/solr"));
+        CLIO.out(
+            "The solrUrl parameter only needs to only point to the root of Solr ("
```

Comment:
[-0] I think this message would read more clearly if you dropped the second "only".

[0] Maybe add some indication in this message that Solr is able to auto-correct the URL for folks, so they don't see the warning and think the command has failed.  e.g.

> 'solrUrl' needn't include Solr's context-root (e.g. "/solr"); correcting from [old] to [new]

Reply:
Love it... pushing up fix.